lib/portlayer/exec/exec.go

```diff
@@ -113,11 +113,14 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 func eventCallback(ie events.Event) {
 	// grab the container from the cache
 	container := Containers.Container(ie.Reference())
+
 	if container != nil {
+		container.m.Lock()
```

Comment:
atomic functions guarantee an atomic read/write as well as that compiler will not cache a value in a register. but it doesn't guarantee a consistency between three different reads/writes.